main.js

```diff
@@ -1,17 +1,37 @@
-const { app, BrowserWindow } = require("electron");
+const { app, BrowserWindow, ipcMain } = require("electron");
+const psList = require("ps-list");
 
 function createWindow() {
     // Create the browser window.
     const win = new BrowserWindow({
         width: 800,
         height: 900,
-        webPreferences: {},
+        webPreferences: {
+            nodeIntegration: true,
+        },
         icon: "tokspace.png",
         title: "TokSpace",
     });
 
     // and load the index.html of the app.
     win.loadURL("http://localhost:3000");
+
+    // Listen for the front-end web app to send a request that asks for what
+    // processes are running on the user's machine.
+    ipcMain.handle("processesRequest", async (_event, ..._args) => {
+        try {
```

Comment:
Please change this entire handler  to just `ipcMain.handle("processesRequest", psList);`

Reply:
Actually, let's do `async () => psList()` to stay on the safe side

Reply:
`psList()` returns a list of objects. Since the `RunningProcesses` component renders each item in the list returned by this handler, I needed to convert each list item to something other than an object. I decided to do that conversion here. Should it be the front-end's job to convert the return value from `psList()` into something that can be rendered, instead?

Reply:
I think it's a good idea to pass along all the information we have, so that the interface can do whatever it would like. One example where this would be useful is Minecraft. The process' name is `java`, but it's hard to identify that it is for Minecraft. We can engineer the UI to provide more information about the command used to run it, perhaps on hover with the `title` attribute.

Reply:
Oh can we also move this out of `createWindow` since it doesn't have anything to do with creating the window

Reply:
Implemented these suggested changes. Lmk what you think